Sources/OktaAnalytics/EventScenario.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+import Foundation
+import Combine
+import CoreData
+
+// Extension to the Date struct, defines the subtraction operator, returning the time interval between two dates
+extension Date {
+    static func - (lhs: Date, rhs: Date) -> TimeInterval {
+        // Returns the time interval between the two dates
+        return lhs.timeIntervalSinceReferenceDate - rhs.timeIntervalSinceReferenceDate
+    }
+}
+
+/// This code defines a Swift class `EventScenario` which represents an event that can be tracked with analytics.
+class EventScenario {
+
+    // eventName is a constant property, initialized with the value passed to the init method
+    private var scenario: ScenarioEvent
+
+    var name: String {
+        scenario.name
+    }
+
+    // properties is a dictionary of strings, initialized as an empty dictionary
+    private(set) var properties: Properties = [:]
+    // time is a property of type Date, initialized as the current date
+    private var time = Date()
+
+    // eventStream is a PassthroughSubject, which is private set
+    private(set) var eventStream: PassthroughSubject<Property, Never>?
```

Comment:
why are we using publisher and subscriber pattern get updates on properties? There is no external functionality that need access publish events ?